test/ServiceTest.cpp

```diff
@@ -408,6 +408,31 @@ TEST_F(ServiceTest, computeResult) {
         siblingTree};
     EXPECT_NO_THROW(serviceOperation6.computeResultOnlyForTesting());
     RuntimeParameters().set<"service-max-value-rows">(maxValueRowsDefault);
+
+    // Check 7: Lazy computation
+    Service lazyService{
+        testQec, parsedServiceClause,
+        getResultFunctionFactory(
+            expectedUrl, expectedSparqlQuery,
+            genJsonResult({"x", "y"},
+                          {{"bla", "bli"}, {"blu", "bla"}, {"bli", "blu"}}),
+            boost::beast::http::status::ok, "application/sparql-results+json")};
+
+    auto lazyResult = lazyService.computeResultOnlyForTesting(true);
+    // Consume the idTable-generator to actually compute the result.
+    for ([[maybe_unused]] auto& _ : lazyResult.idTables()) {
+    }
```

Comment:
You should of course check the result. You can for example initialize an IdTable, and append all the partial results via `insertAtEnd`.